docs/reference/sqrl/connectors.md

```diff
@@ -0,0 +1,246 @@
+
+The table configuration has three sections for defining the table properties, connector configuration, and - optionally - metadata columns.
+
+An example table configuration file is shown below followed by the documentation for each section.
+
+```json
+{
+  "version": 1,
+  "table" : {
+    "type" : "source",
+    "primary-key" : ["id", "time"],
+    "timestamp" : "_source_time",
+    "watermark-millis" : "0"
+  },
+  "flink" : {
+    "format" : "avro",
+    "bootstrap.servers": "${BOOTSTRAP_SERVERS}",
+    "group.id": "datasqrl-orders",
+    "connector" : "kafka"
+  },
+  "metadata" : {
+    "_source_time" : {
+      "attribute" : "timestamp",
+      "type": "TIMESTAMP_WITH_LOCAL_TIME_ZONE(3)"
+    }
+  }
+}
+```
+
+### Table Properties
+
+The `table` section of the table configuration specifies the table properties that are used by the DataSQRL planner to validate SQRL scripts and generate efficient data pipelines.
+
+The `type` of a table can be either `source`, `sink`, or `source_and_sink` if it can be used as both.<br />
+
+The `primary-key` specifies the column or list of columns that uniquely identifies a single record in the table. Note, that when the table is a changelog or CDC stream for an entity table, the primary key should uniquely identify each record in the stream and not the underlying table. For example, if you consume a CDC stream for a `Customer` entity table with primary key `customerid` the primary key for the resulting CDC stream should include the timestamp of the change, e.g. `[customerid, lastUpdated]`.
+
+The `timestamp` field specifies the (single) timestamp column for a source stream which has the event time of a single stream record. `watermark-millis` defines the number of milliseconds that events/records can arrive late for consistent processing. Set this to `1` if events are perfectly ordered in time and to `0` if the timestamp is monotonically increasing (i.e. it's perfectly ordered and no two events have the same timestamp). <br />
+Alternatively, you can also use processing time for event processing by removing the `watermark-millis` field and adding the processing time as metadata (see below), which means using the system clock of the machine processing the data and not the timestamp of the record. We highly recommend you use event time and not processing time for consistent, reproducible results. <br />
+Timestamp and watermark are only used for sources.
+
+### Connector Configuration
+
+The connector configuration specifies how the stream engine connects to the source or sink and how it reads or writes the data. The connector configuration is specific to the configured stream processing engine that DataSQRL compiles to and the section of the configuration is named after the engine. In the example above, the connector configuration is for the `flink` engine.
```

Comment:
really needs a comma after the 'to', hard to read when it ends with a preposition.

The connector configuration is specific to the configured stream processing engine that DataSQRL compiles to, and the section of the configuration is named after the engine.